source/slangc/main.cpp

```diff
@@ -43,6 +40,9 @@ static SlangResult innerMain(int argc, char** argv)
 
     spSetCommandLineCompilerMode(compileRequest);
 
+    // Do any app specific configuration
+    appContext->configureRequest(compileRequest);
```

Comment:
Wait... I must have missed the part where `AppContext` in the `core` library has a dependency on `SlangCompileRequest`. That seems to seriously invert the relationships that are supposed to exist between the components of the system.

Reply:
Well this is the 'injection' part of how when invoked from the dll, we have a way of altering the behavior. In this case applying the streams. 
I'm not keen on how this turned out, when I started it wasn't clear what state was needed/injection. Anyway this can and probably should be improved.